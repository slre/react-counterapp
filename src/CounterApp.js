import React, { useState } from 'react';
import PropTyes from 'prop-types';

const CounterApp = ( { value = 105 } ) => {

    //const [ name, setName ] = useState('Goku');
    //console.log(name,setName);
    const [ counter,setCounter ] = useState( value );
    

    const handleAdd = () => {
       // setCounter (counter+1 );
        setCounter ( (c)=>c+1 );
    }
    const handleSubs = (c) => {
         setCounter ( (c)=>c-1 )
    }
     const handleReset = () => {
        setCounter ( value );
    } 
    return ( 
        <>
            <h1> Counter App</h1>
            <h2>{ counter }</h2>
            {/* <button onClick={ (e) =>{ handleAdd(e) } } >+1</button> */}
            {/*
            handleAdd references the 'e' parameter
            */}
            <button onClick={  handleAdd } >+1</button>
            <button onClick={  handleSubs } >-1</button>
            <button onClick={  handleReset } >reset</button>
        </>

    );
}

CounterApp.propTypes = {
    value:PropTyes.number
}
// CounterApp.defaultProps = {
//     value:100
// }
export default CounterApp;