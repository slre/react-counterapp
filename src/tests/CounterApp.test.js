import React from 'react';
import { render } from '@testing-library/react';
import CounterApp from '../CounterApp';
import { shallow } from 'enzyme';
import  '@testing-library/jest-dom'; 

describe('Testing <CounterApp>', () => { 

  //let wrapper; // UNDEFINED: if not initialized, 'looses' intelligence from editor
  let wrapper = shallow(<CounterApp  />); // Keep intelligence editof functionallity 
  beforeEach( ()=>{
    
    wrapper = shallow(<CounterApp  />);

  })

  test('Should show <CounterApp /> correctly', () => {
    // const wrapper = shallow( 
    //   <CounterApp  
    //   />);
    expect( wrapper ).toMatchSnapshot();
  });
  test('Should show <CounterApp /> correctly', () => {
    const value = 100;
    const wrapper = shallow( 
      <CounterApp  
        value={ value }
      />);
    const defautlVaue = wrapper.find('h2').text();
     expect( defautlVaue ).toBe( value.toString() );
  });
  test('Increase 1', () => { 
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text();
    expect( counterText ).toBe('11');
   })
   test('Decrease 1', () => { 
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text();
    expect( counterText ).toBe('9');
   })
   test('Reset to default value at clicking reset', () => { 
    const wrapper = shallow( <CounterApp  value = { 105 } />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text();
    expect( counterText ).toBe('105');
    
   })
 })