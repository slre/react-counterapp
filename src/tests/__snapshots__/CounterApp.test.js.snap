// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CounterApp> Should show <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
     Counter App
  </h1>
  <h2>
    105
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
